symbol-processing-cmdline/build.gradle.kts

```diff
@@ -0,0 +1,57 @@
+import com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar
+
+evaluationDependsOn(":common-util")
+evaluationDependsOn(":compiler-plugin")
+
+val kotlinBaseVersion: String by project
+val signingKey: String? by project
+val signingPassword: String? by project
+
+plugins {
+    kotlin("jvm")
+    id("com.github.johnrengelman.shadow") version "6.0.0"
+    `maven-publish`
+    signing
+}
+
+val packedJars by configurations.creating
+
+dependencies {
+    packedJars(project(":compiler-plugin")) { isTransitive = false }
+    packedJars(project(":common-util")) { isTransitive = false }
+}
+
+tasks.withType<ShadowJar>() {
+    archiveClassifier.set("")
+    // ShadowJar picks up the `compile` configuration by default and pulls stdlib in.
+    // Therefore, specifying another configuration instead.
+    configurations = listOf(packedJars)
+}
+
+tasks {
+    publish {
+        dependsOn(shadowJar)
+        dependsOn(project(":compiler-plugin").tasks["dokkaJavadocJar"])
+        dependsOn(project(":compiler-plugin").tasks["sourcesJar"])
+        dependsOn(project(":common-util").tasks["dokkaJavadocJar"])
+        dependsOn(project(":common-util").tasks["sourcesJar"])
+    }
+}
+
+publishing {
+    publications {
+        create<MavenPublication>("shadow") {
+            artifactId = "symbol-processing-cmdline"
+            artifact(tasks["shadowJar"])
+            pom {
+                name.set("com.google.devtools.ksp:symbol-processing-cmdline")
+                description.set("Symbol processing for K/N and command line")
```

Comment:
These 2 lines are not enough. ShadowJar missed the pom file. You need to either create the dependencies manually lke [this](https://github.com/google/ksp/blob/main/symbol-processing/build.gradle.kts#L47), or figure out how to make ShadowJar copy over the POM file.

Reply:
For reference, this is the diff before and after the change in the POM:
```
-  <dependencies>
-    <dependency>
-      <groupId>org.jetbrains.kotlin</groupId>
-      <artifactId>kotlin-stdlib</artifactId>
-      <version>1.7.0-dev-1132</version>
-      <scope>runtime</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.jetbrains.kotlin</groupId>
-      <artifactId>kotlin-compiler</artifactId>
-      <version>1.7.0-dev-1132</version>
-      <scope>runtime</scope>
-    </dependency>
-    <dependency>
-      <groupId>com.google.devtools.ksp</groupId>
-      <artifactId>symbol-processing-api</artifactId>
-      <version>2.0.0</version>
-      <scope>runtime</scope>
-    </dependency>
-  </dependencies>
```

Reply:
it was copied from old `symbol-processing-cmdline` publication task, the pom was missing before because in K/N and cmd use case, pom is not needed, I followed the behavior, I can add it back if necessary.

Reply:
Thanks for adding it. So the interesting thing is that `symbol-processing-cmdline` WAS produced directly by the maven publish plugin, which produces POM automatically. It's ShadowJar, which we are using now, doesn't copy the POM. Also, although cmd line doesn't need it, K/N in Gradle does require it.